hs_restclient/endpoints/resources.py

```diff
@@ -0,0 +1,237 @@
+            folder: folder path to upload the file to
+        :return: json object
+            resource_id: string resource id,
+            file_name: string name of file
+        """
+        url = "{url_base}/resource/{pid}/files/".format(url_base=self.hs.url_base,
+                                                       pid=self.pid)
+
+        params = {}
+        file_to_upload = self.hs._prepareFileForUpload(params,
+                                                       payload['file'],
+                                                       os.path.basename(payload['file']))
+
+        encoder = MultipartEncoder(params)
+        monitor = MultipartEncoderMonitor(encoder, default_progress_callback)
+
+        payload.pop('file')
+        r = self.hs._request('POST', url, None, data=monitor, headers={'Content-Type': monitor.content_type})
+        return r.text
+
+    def version(self):
+        """
+        Creates a new version of a resource
+
+        :return: resource id (string)
+        """
+        url = "{url_base}/resource/{pid}/version/".format(url_base=self.hs.url_base,
+                                                          pid=self.pid)
+        r = self.hs._request('POST', url)
+        return r
+
+
+class ResourceList(BaseEndpoint):
+    def __init__(self, hs, **kwargs):
+        super(ResourceList, self).__init__(hs)
+
+        """
+        Query the GET /hsapi/resource/ REST end point of the HydroShare server.
+
+        :param creator: Filter results by the HydroShare user name of resource creators
+        :param owner: Filter results by the HydroShare user name of resource owners
+        :param user: Filter results by the HydroShare user name of resource users (i.e. owner, editor, viewer, public
+            resource)
+        :param group: Filter results by the HydroShare group name associated with resources
+        :param from_date: Filter results to those created after from_date.  Must be datetime.date.
+        :param to_date: Filter results to those created before to_date.  Must be datetime.date.  Because dates have
+            no time information, you must specify date+1 day to get results for date (e.g. use 2015-05-06 to get
+            resources created up to and including 2015-05-05)
+        :param types: Filter results to particular HydroShare resource types.  Must be a sequence type
+            (e.g. list, tuple, etc.), but not a string.
+
+        :raises: HydroShareHTTPException to signal an HTTP error
+        :raises: HydroShareArgumentException for any invalid arguments
+
+        :return: A generator that can be used to fetch dict objects, each dict representing
+            the JSON object representation of the resource returned by the REST end point.  For example:
+
+        >>> for resource in hs.getResourceList():
+        >>>>    print resource
+         {u'bag_url': u'http://www.hydroshare.org/static/media/bags/e62a438bec384087b6c00ddcd1b6475a.zip',

```

Comment:
Seems the return json data for resource is missing few attributes/keys (https://github.com/hydroshare/hydroshare/blob/develop/hs_core/views/serializers.py#L91)